Size component setup

nathanq95 opened the pull request:
Set up size component to render based on the set of sizes available for a given product/style.  Also added a hover event for the size grid.  I installed additional dependencies to resolve a cors issue.

martinsung00 commented on client/sizeGrid.jsx:
```diff
@@ -0,0 +1,71 @@
+    this.getSizes()
+    .then((results) => {
+      this.setState({sizes: results});  
+      console.log(this.state.sizes);
```
Remember to remove any unnecessary console logs; if this is still needed for future development then keep it, otherwise you should remove it.

martinsung00 commented on client/sizeGrid.jsx:
```diff
@@ -0,0 +1,71 @@
+      console.log(this.state.sizes);
+    })
+    .catch((err) => {
+      console.log('ERROR FETCHING STATE: ', err);
```
Consider using a more formal way of displaying errors, such as 'throw (error)'

martinsung00 commented on client/sizeGrid.jsx:
```diff
@@ -0,0 +1,71 @@
+  }
+
+  selectSize (selectedSize) {
+    let selected = document.getElementsByClassName(selectedSize)[0];
```
Since selected is not going to change, you should use const here.

martinsung00 commented on client/sizeGrid.jsx:
```diff
@@ -0,0 +1,71 @@
+  }
+}
+
+let domContainer = document.querySelector('#app');
```
Dom container is also a constant variable that won't change anytime soon so you can safely use const here.

martinsung00 reviewed (approved):
Some 'let' can be altered to 'const', aside from that the css/styles are very organized. No issues with the functions itself, clean, concise and easy to understand. Remember to remove unnecessary console logs, other than that, this PR is approved.

nathanq95 commented:
@martinsung00 Per your instruction, I changed the variables to use const, removed the console.logs, and updated my then statement to throw the error.  In addition, I added the functionality to display if a style is almost sold out.  I also added the add to bag & favorite buttons and  the shipping text.  Lastly, I reconfigured webpack to bundle the size component by itself.
